superSet Documentation - mh (#1266)
Codecademy/docs: content/swift/concepts/sets/terms/isSuperset/isSuperset.md

---
Title: '.isSuperset()'

> Description: 'Checks if a set contains every element of another set.'

Reviewer: ```suggestion
Description: 'Returns a boolean telling whether every element of a given set exists in another set.'
```

Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Sets'
  - 'Methods'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **`.isSuperset()`** method returns `true` if a set contains every element that exists in another set (passed as argument), `false` otherwise.

Reviewer: ```suggestion
The **`.isSuperset()`** method returns a boolean telling whether every element of a given set exists in another set.
```

## Syntax

```pseudo
setOne.isSuperset(of: setTwo)
```

> The `isSuperset()` method returns `true` if `setOne` is a superset of `setTwo`, `false` otherwise.

Reviewer: ```suggestion
The `.isSuperset()` method returns `true` if `setOne` is a superset of `setTwo`, `false` otherwise.
```

## Example

> 

Reviewer: ```suggestion

In the following example, the `.isSuperset()` method is used with a [conditional expression](https://www.codecademy.com/resources/docs/swift/conditionals) to print a phrase:

```

```swift
let setOne: Set = ["Apple", "Orange", "Banana", "Kiwi", "Tomato"]
let setTwo: Set = ["Orange", "Banana"]

if setOne.isSuperset(of: setTwo) {
  print("setOne is a superset of setTwo.")
} else {
  print("setOne is not a superset of setTwo.")
}
```

> In the example above, the first block after the `if` statement will run if `setOne` is a superset of `setTwo`. This will output:

Reviewer: ```suggestion
This will output the following:
```

Author: That's more concise. Thank you

---
Title: '.isSuperset()'
Description: 'Checks if a set contains every element of another set.'
Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Sets'
  - 'Methods'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

The **`.isSuperset()`** method returns `true` if a set contains every element that exists in another set (passed as argument), `false` otherwise.

## Syntax

```pseudo
setOne.isSuperset(of: setTwo)
```

The `isSuperset()` method returns `true` if `setOne` is a superset of `setTwo`, `false` otherwise.

## Example

```swift
let setOne: Set = ["Apple", "Orange", "Banana", "Kiwi", "Tomato"]
let setTwo: Set = ["Orange", "Banana"]

if setOne.isSuperset(of: setTwo) {
  print("setOne is a superset of setTwo.")
} else {
  print("setOne is not a superset of setTwo.")
}
```

> In the example above, the first block after the `if` statement will run if `setOne` is a superset of `setTwo`. This will output:

Reviewer 2: We should only have the actual output of the code in the example. Including the hypothetical content is potentially confusing to a novice.

Author: > We should only have the actual output of the code in the example. Including the hypothetical content is potentially confusing to a novice.

That makes sense. I'll get that changed.
